sql/wsrep_mysqld.cc

```diff
@@ -2583,3 +2583,17 @@ const char* wsrep_get_wsrep_status(wsrep_status status)
 
   return "NULL";
 }
+
+/*
+ Returns non-const copy of thd->rewritten_query().
+
+ PXC code uses rewritten_query.c_ptr_safe() in several places.
+ As c_ptr_save() is non const method, it requires non const object.
+ Here we return copy of thd->rewritten_query() instead of const-casting
+ thd->rewritten_query() in place.
```

Comment:
This could also possibly result in a reallocation in a temporary object, after which we will end up with a dangling pointer. 

Are you sure this is correct, have you tried an ASAN test run too?

Reply:
As we discussed on Slack, all functions that use pointer from the non const copy of string are fprintf-like, so there is no possibility of store the pointer somewhere and use it later.